hazelcast/protocol/builtin.py

```diff
@@ -535,6 +539,31 @@ def decode(msg):
         return result
 
 
+class FloatArrayCodec:
```

Comment:
not related to this pr but I realized most of the code is repeating in this builtin codecs, i wonder why we don't reuse code.

Reply:
I guess that would be a separate tech debt cleanup task.

Reply:
yeah asking because you may know python stuff more: for example maybe reusing code is costly due to method extra invocations etc. 🤷🏻‍♂️